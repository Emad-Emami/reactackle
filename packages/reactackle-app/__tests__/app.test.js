'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import {
    App,
    BottomRegion,
    TopRegion,
    MainRegion,
    Content,
} from '../src';

jest.mock('react-dom');

describe('<App/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <App />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop fixed set true', () => {
    const tree = renderer.create(
      <App fixed />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop fixed set false', () => {
    const tree = renderer.create(
      <App fixed={false} />,
    );
    
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if contains children', () => {
    const tree = renderer.create(
      <App>
        <TopRegion />
        <MainRegion>
          <Content>
            <div>TEST</div>
          </Content>
        </MainRegion>
        <BottomRegion />
      </App>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  
  it('renders correctly if prop fixed changing', () => {
    const component = renderer.create(
      <App>
        <MainRegion>
          <Content>
            <div>TEST</div>
          </Content>
        </MainRegion>
      </App>,
    );
    
    expect(component.toJSON()).toMatchSnapshot();
    
    const instance = component.getInstance();
    instance.componentWillReceiveProps({ fixed: true });
    
    expect(component.toJSON()).toMatchSnapshot();
  });
});
