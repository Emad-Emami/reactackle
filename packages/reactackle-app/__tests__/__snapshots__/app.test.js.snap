// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders correctly if contains children 1`] = `
<div
  className="AppStyled-jBxgUA iVjDIN"
>
  <div
    className="TopRegionStyled-cVQlko gRcViP"
  >
    <div
      className="TopRegionContentStyled-iFwSwY ewMYrE"
    />
  </div>
  <div
    className="MainRegionStyled-gqcCET ePyrOs"
  >
    <div
      className="ContentStyled-dbeIsI gAKDxp"
    >
      <div>
        TEST
      </div>
    </div>
  </div>
  <div
    className="BottomRegionStyled-djrNnR efysWV"
  >
    <div
      className="BottomRegionContentStyled-eNFvSp igfcXw"
    />
  </div>
</div>
`;

exports[`<App/> renders correctly if prop fixed changing 1`] = `
<div
  className="AppStyled-jBxgUA iVjDIN"
>
  <div
    className="MainRegionStyled-gqcCET ePyrOs"
  >
    <div
      className="ContentStyled-dbeIsI gAKDxp"
    >
      <div>
        TEST
      </div>
    </div>
  </div>
</div>
`;

exports[`<App/> renders correctly if prop fixed changing 2`] = `
<div
  className="AppStyled-jBxgUA iVjDIN"
>
  <div
    className="MainRegionStyled-gqcCET diTFYw"
  >
    <div
      className="ContentStyled-dbeIsI gdEPoZ"
    >
      <div>
        TEST
      </div>
    </div>
  </div>
</div>
`;

exports[`<App/> renders correctly if prop fixed set false 1`] = `
<div
  className="AppStyled-jBxgUA iVjDIN"
/>
`;

exports[`<App/> renders correctly if prop fixed set true 1`] = `
<div
  className="AppStyled-jBxgUA kzApsw"
/>
`;

exports[`<App/> renders correctly with default props 1`] = `
<AppStyled
  fixed={false}
/>
`;
